std/math/powi.zig

```diff
@@ -0,0 +1,137 @@
+// Special Cases:
+//
+//  powi(x, +-0)    = 1 for any x
+//  powi(1, y)      = 1 for any y
+//  powi(x, 1)      = x for any x
+//  powi(nan, y)    = nan
```

Comment:
is there a nan special case for integers? Looks like maybe all these special case comments are from floats and should be deleted.